### PR TITLE
Add a `YkMt*` first argument to control points.

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: llvm/lib/Transforms/Yk/ControlPoint.cpp [hunk 1]
```diff
@@ -59,6 +61,11 @@
 #define DEBUG_TYPE "yk-control-point"
 #define JIT_STATE_PREFIX "jit-state: "
 
+#define YK_CONTROL_POINT_MT_IDX 0
+#define YK_CONTROL_POINT_LOC_IDX 1
```
Comment (vext01) [inline llvm/lib/Transforms/Yk/ControlPoint.cpp:65]: Should we consistently include `ARG` in the constant name for argument indices?

File: llvm/lib/Transforms/Yk/ControlPoint.cpp [hunk 2]
```diff
@@ -156,11 +163,16 @@ class YkControlPoint : public ModulePass {
     StructType *CtrlPointVarsTy =
         StructType::create(TypeParams, "YkCtrlPointVars");
 
+    // The old control point should be of the form:
+    //    control_point(YkMT*, YkLocation*)
+    assert(OldCtrlPointCall->arg_size() == 2);
```
Comment (vext01) [inline llvm/lib/Transforms/Yk/ControlPoint.cpp:168]: For consistency, should `2` be a named constant too?

**Comment by vext01** [comment]
LGTM, bar minor comments.

File: llvm/lib/Transforms/Yk/ControlPoint.cpp (hunk 1 above)
Comment (ltratt) [reply llvm/lib/Transforms/Yk/ControlPoint.cpp:65]: Agreed. Fixed in 07bc55a798db.

**Comment by ltratt** [comment]
@vext01 Hopefully that addresses the comments!

**Comment by vext01** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**